cmd/installOperator.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"github.com/operate-first/opfcli/api"
+	"github.com/operate-first/opfcli/constants"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdInstallOperator(opfapi *api.API) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install-operator name catalog",
+		Short: "Install an operator",
+		Long: `Install an operator via Operator Hub.
+
+- Register a new subscription
+- Register a new operator group and namespace if requested
+`,
+		Args:          cobra.ExactArgs(2),
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			name := args[0]
+			source := args[1]
+			namespace, err := cmd.Flags().GetString("namespace")
+			if err != nil {
+				return err
+			}
+			scope, err := cmd.Flags().GetBool("scope-to-single-namespace")
+			if err != nil {
+				return err
+			}
+			channel, err := cmd.Flags().GetString("channel")
+			if err != nil {
+				return err
+			}
+			manual, err := cmd.Flags().GetBool("manual")
+			if err != nil {
+				return err
+			}
+			owner, err := cmd.Flags().GetString("owner")
+			if err != nil {
+				return err
+			}
+
+			if err := opfapi.CreateNamespace(namespace, owner, "", "", true, true); err != nil {
+				return err
+			}
+
+			if err := opfapi.CreateOperatorGroup(namespace, !scope); err != nil {
+				return err
+			}
+
+			return opfapi.CreateSubscription(
+				name, source, namespace, channel, manual,
+			)
+		},
+	}
+
+	cmd.Flags().StringP("namespace", "n", constants.DefaultOperatorNamespace, "Namespace where the operator will be deployed to")
+	cmd.Flags().StringP("owner", "o", constants.DefaultOwner, "If a new namespace is created, owner must be set as well")
+	cmd.Flags().StringP("channel", "c", "", "Subscription channel")
+	cmd.Flags().BoolP("manual", "m", false, "Require manual approval install strategy")
+	cmd.Flags().BoolP("scope-to-single-namespace", "s", false, "Set an operator scope to a single namespace")
```

Comment:
Could we maybe just call this `--single-namespace` instead of `--scope-to-single-namespace`?

Reply:
Hm.. I've tried to differentiate the namespace where the operator gets deploy to and the scope operators is able to manage - naming both `*namespace` seemed too confusing to me. any suggestion?

What about I do a string type flag and allow `AllNamespaces` or `OwnNamespace` values only (matching the OLM naming conventions)?

Reply:
But you've *already* named them both `*namespace`. My suggestion was just to shorten `--scope-to-single-namespace` down to `--single-namespace`, because it's a flag, not a sentence.

Reply:
Ok.. I'll come up with a better name then. :smile:

Reply:
What about now? It should match OLM terminology better: 
https://docs.openshift.com/container-platform/4.7/operators/understanding/olm/olm-understanding-operatorgroups.html#olm-operatorgroups-membership_olm-understanding-operatorgroups

Reply:
Sounds reasonable, ok I can revert it. No big deal. :slightly_smiling_face:  Changed

Reply:
I had to rebase anyways :smile: